processing/src/main/java/io/druid/query/filter/LikeDimFilter.java

```diff
@@ -75,94 +79,113 @@ public LikeDimFilter(
       MATCH_PATTERN
     }
 
+    // Strings match if they start with "prefix" AND:
```

Comment:
Maybe these comments could be Javadocs of `SuffixStyle` enum or constants themselves?